src/reducers.js

```diff
@@ -63,23 +64,37 @@ const perPage = (state = PER_PAGE) => {
   return state
 }
 
-// const errorReport = (state = '', action) => {
-//   switch (action.type) {
-//     case REPORT_ERROR:
-//       console.log(`${action.errorCode}: ${action.errorMessage}`)
-//       return `${action.errorCode}: ${action.errorMessage}`
-//     default:
-//       return state
-//   }
-
-// }
-
-const rootReducer = combineReducers({
+const appReducer = combineReducers({
   userData,
   currentPage,
   allUsers,
   pageUsers,
   perPage
 })
 
+// Async actions
+// ---------------------------------------------------------------------------
+const login = (email, password) => {
+  return firebase.auth().signInWithEmailAndPassword(email, password)
+    .then((user) => {
+      firebase.auth().onAuthStateChanged((fbuser) => {
+        return fbuser ? fbuser.uid : null
```

Comment:
This is the code I pulled out of `Main.js`. I also cleaned it up a little. Specifically, 
```js
if (fbuser) {
  userUID = fbuser.uid
  this.props.actions.setAUser(userUID)
} else {
  userUID = null
  this.props.actions.setAUser(userUID)
}
```
becomes simply this line. A redux-loop action needs to return a promise for a result, and that is all. We will therefore simply return the `fbuser.uid`

Reply:
In fact, once this sinks in a little, I would encourage you to convert this to an `async` function and see how much cleaner and simpler the code gets.